pip/operations/prepare.py

```diff
@@ -149,195 +149,219 @@ def _download_should_save(self):
         return False
 
     def prepare_requirement(self, req, resolver):
-        # TODO: Breakup into smaller functions
-        # TODO: Add a nice docstring
+        """Prepare a requirement for installation
+
+        Returns an AbstractDist that can be used to install the package
+        """
+        # TODO: Remove circular dependency on resolver
```

Comment:
I think I'll do this by actually moving parts of this flow into `resolver.resolve`